packages/backend/src/plugins/catalog.ts

```diff
@@ -15,12 +15,28 @@
  */
 
 import {
-  DatabaseCatalog,
+  DatabaseItemsCatalog,
+  DatabaseLocationsCatalog,
+  DatabaseManager,
   createRouter,
+  runPeriodically,
+  LocationReaders,
+  DescriptorParsers,
 } from '@backstage/plugin-catalog-backend';
 import { PluginEnvironment } from '../types';
 
-export default async function({ logger, database }: PluginEnvironment) {
-  const catalog = await DatabaseCatalog.create(database, logger);
-  return await createRouter({ catalog, logger });
+export default async function ({ logger, database }: PluginEnvironment) {
+  const reader = LocationReaders.create();
+  const parser = DescriptorParsers.create();
+
+  const db = await DatabaseManager.createDatabase(database);
+  runPeriodically(
```

Comment:
Thinking this likely needs to be a bit more encapsulated. It'll likely be a bit tedious with a bunch of different database implementations with different methods of keeping up to date. Maybe just let `createDatabase` or a variant of that start the loop, and add a `.close` method to the DB?

Reply:
A database wouldn't know that there's a refresh loop going on since it's at a higher level. Anyway, it's left in this pretty basic state for now because there's a ticket coming up to change the method of refresh to instead be based on "last refresh attempt timestamp" per location etc

Reply:
Hm and I hope that this will be THE implementation either way, not one of many. Which on the other hand does mean that you are correct that it could be more encapsulated. Anyway, TODO